### PR TITLE
Add read_from_socket function & add serialize method to JobSocket

### PR DESCRIPTION
**Pull request by JanusAsmussen**

### CODE REVIEW

**Review by george-zubrienko — COMMENTED** [review]
LGTM, but needs units tests for both methods. Check `tests` for inspiration

**Comment by JanusAsmussen** [comment]
> LGTM, but needs units tests for both methods. Check `tests` for inspiration

Added some tests now

File: test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import pytest
+import pathlib
+
+from spark_utils.common.spark_session_provider import SparkSessionProvider
+
+
+@pytest.fixture(scope="module")
+def spark_session():
+    provider = SparkSessionProvider()
+    return provider.get_session()
```
Comment (george-zubrienko) [inline test/conftest.py:10]:
```suggestion
    return SparkSessionProvider().get_session()
```

File: test/test_common_functions/data.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"strings":{"0":"abc","1":"def","2":"ghe"},"ints":{"0":1,"1":2,"2":3},"floats":{"0":1.0,"1":2.0,"2":3.0}}
```
Comment (george-zubrienko) [inline test/test_common_functions/data.json:1]:
```suggestion
{"strings":{"0":"abc","1":"def","2":"ghe"},"ints":{"0":1,"1":2,"2":3},"floats":{"0":1.0,"1":2.0,"2":3.0}}

```

**Review by george-zubrienko — APPROVED** [review]
LGTM, left some style-related comments. Thanks for test refactor :)

File: src/spark_utils/common/functions.py [hunk 3]
```diff
@@ -57,3 +60,17 @@ def decrypt_sensitive(sensitive_content: Optional[str]) -> Optional[str]:
         return fernet.decrypt(sensitive_content.encode('utf-8')).decode('utf-8')
 
     return None
+
+
+def read_from_socket(socket: JobSocket, spark_session: SparkSession, **kwargs) -> DataFrame:
```
[the remark this answers is not in the document]
Comment (JanusAsmussen) [reply src/spark_utils/common/functions.py:65]: Good point, fixed

**Pull request merged**